src/ralph/data_center/models/virtual.py

```diff
@@ -26,13 +27,30 @@ def __str__(self):
         return 'Database: {}'.format(self.service_env)
 
 
+class VIPProtocol(Choices):
+    _ = Choices.Choice
+
+    unknown = _('unknown')
+    TCP = _('TCP')
+    HTTP = _('HTTP')
+
+
 class VIP(AdminAbsoluteUrlMixin, BaseObject):
     class Meta:
         verbose_name = _('VIP')
         verbose_name_plural = _('VIPs')
+        unique_together = ('ip', 'port', 'protocol')
 
     def __str__(self):
-        return 'VIP: {}'.format(self.service_env)
+        return self.name
+
+    name = models.CharField(_('name'), max_length=255, blank=True, null=True)
+    ip = models.ForeignKey(IPAddress, blank=True, null=True, default=None)
```

Comment:
how could this be blank or null? doesn't every vip has defined ip (and port and protocol)?